.github/workflows/minimum_dependency_checker.yaml

```diff
@@ -0,0 +1,64 @@
+name: Minimum dependency checker
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - 'pyproject.toml'
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+      - name: Run min dep generator - test requirements
+        id: min_dep_gen_test
+        uses: alteryx/minimum-dependency-generator@v3.4
+        with:
+          paths: 'pyproject.toml'
+          options: 'dependencies'
+          extras_require: 'test'
+      - name: Save min test requirements and run diff
+        id: check_min_test
+        continue-on-error: true
+        run: |
+          mkdir /tmp/dependencies_updated_artifacts
+          printf "${{ steps.min_dep_gen_test.outputs.min_reqs }}" > /tmp/minimum_test_requirements.txt
+          diff /tmp/minimum_test_requirements.txt tests/dependency_update_check/minimum_test_requirements.txt
+      - name: if min test requirements have changed, write output file
+        if: steps.check_min_test.outcome != 'success'
+        run: |
+          printf "${{ steps.min_dep_gen_test.outputs.min_reqs }}" > tests/dependency_update_check/minimum_test_requirements.txt
+      - name: Run min dep generator - requirements
+        id: min_dep_gen_reqs
+        uses: alteryx/minimum-dependency-generator@v3.4
+        with:
+          paths: 'pyproject.toml'
+          options: 'dependencies'
+      - name: Save min requirements and run diff
+        id: check_min_reqs
+        continue-on-error: true
+        run: |
+          printf "${{ steps.min_dep_gen_reqs.outputs.min_reqs }}" > /tmp/minimum_requirements.txt
+          diff /tmp/minimum_requirements.txt tests/dependency_update_check/minimum_requirements.txt
+      - name: if min requirements have changed, write output file
+        if: steps.check_min_reqs.outcome != 'success'
+        run: |
+          printf "${{ steps.min_dep_gen_reqs.outputs.min_reqs }}" > tests/dependency_update_check/minimum_requirements.txt
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@v3
+        with:
+          token: ${{ secrets.MACHINEFL_DEPENDENCY_CHECKER_TOKEN }}
```

Comment:
Have all the relevant tokens been set already?

Reply:
No tokens have been set yet as I'm still working on this proj in my personal repo, I believe once it gets transferred over to AYX, this will be set up? Unsure how to proceed however.

Reply:
Yep sounds good we can set it up later